### PR TITLE
Fixes for Brightcove Player and videojs integration

### PR DESCRIPTION
**Pull request by tjaneczko**
A number of fixes were made to the YouTube tech to allow it to be used in the Brightcove Player.  Most of these fixes also fix issues which could come up using a videojs player with certain plugins.  Here's a list of everything that was changed:

* Added .idea to the `.gitignore` file to prevent WebStorm files from being added to git
* Ensure the version of videojs we're using has the `getComponent` function before loading the tech
* Added the `vjs-tech` class to the tech element div.  This is used in several plugins with `querySelector` to retrieve the tech element div and is needed when used in a Brightcove Player.
* Added video cueing when switching sources with autoplay disabled.
* Fixed `onPlayerStateChange` by setting `laststate` before triggering any events as some listeners use the state of the player (which is stored in `laststate`)
* Added the `loadstart` event to the loading state (state -1)
* Added required videojs tech functions `autoplay`, `setAutoplay`, `loop`, `seeking`, `seekable`, and `preload`
* Return the actual source url and not the source object in `currentSrc`
* Don't use native controls on Android browsers that support using HTML controls
* Fixed conflicts with other plugins (such as IMA) which load YouTube assets by changing the loading of the iframe API.  Now, we wait for the API script to be loaded, then we register a callback to the `YT.ready` function instead of setting a `window. onYouTubeIframeAPIReady` function.  The `YT.ready` callback function is run once the API is ready to instantiate players and prevents timing issues where the `onYouTubeIframeAPIReady` function could potentially be overwritten by other code.

Feel free to leave any comments or ask any questions about any of the changes / fixes.  I am going to be away on vacation for the next week, but should hopefully be able to get around to responding to them during that time.

### CODE REVIEW

File: src/Youtube.js [hunk 1]
```diff
@@ -339,9 +345,28 @@ THE SOFTWARE. */
...
 
+    autoplay: function() {
+      return this.options_.autoplay;
+    },
+
+    setAutoplay: function(val) {
+      this.options_.autoplay = val;
+    },
+
+    loop: function() {
```
Comment (gkatsev) [inline src/Youtube.js]: does loop not get a setter?

File: src/Youtube.js [hunk 2]
```diff
@@ -33,6 +33,10 @@ THE SOFTWARE. */
 }(this, function(videojs) {
   'use strict';
 
+  if (!videojs.getComponent) {
```
Comment (gkatsev) [inline src/Youtube.js]: Why not just assume it will only be used with videojs 5 and above?

File: src/Youtube.js [hunk 3]
```diff
@@ -291,10 +300,6 @@ THE SOFTWARE. */
     src: function(src) {
       if (src) {
         this.setSrc({ src: src });
-
-        if (this.options_.autoplay && !_isOnMobile) {
```
Comment (gkatsev) [inline src/Youtube.js]: why remove this?

Comment (tjaneczko) [inline src/Youtube.js]: This is already handled in `this.setSrc`, `this.play()` was already being called twice in the tech.

File: src/Youtube.js (hunk 2 above)
Comment (tjaneczko) [inline src/Youtube.js]: We actually should be doing that considering there's a note in the README about using videojs 5 with the tech, going to remove this check.

File: src/Youtube.js (hunk 1 above)
Comment (tjaneczko) [inline src/Youtube.js]: Added a setLoop function to the tech.

File: src/Youtube.js (hunk 3 above)
Comment (gkatsev) [inline src/Youtube.js]: I wish github had a "resolved" button for comment threads.

**Comment by gkatsev** [comment]
LGTM.

**Pull request merged**